### PR TITLE
don't panic on 4byte IP addresses

### PR DESCRIPTION
**Pull request by Stebalien**
They're valid.

### CODE REVIEW

File: net/net.go [hunk 1]
```diff
@@ -115,22 +115,26 @@ func NetAddrToSockaddr(addr net.Addr) Sockaddr {
...
 func IPAndZoneToSockaddr(ip net.IP, zone string) Sockaddr {
-	switch {
-	case len(ip) < net.IPv4len: // default to IPv4
-		buf := [4]byte{0, 0, 0, 0}
-		return &SockaddrInet4{Addr: buf}
-
-	case ip.To4() != nil:
+	if zone != "" {
+		var buf [16]byte
+		copy(buf[:], ip.To16())
```
Comment (vyzo) [inline net/net.go]: So, if `ip.To16` returns nil, we'll end up with the unspecific address. 
Is this desirable? Or is the assumption that the user wouldn't pass a zone unless they had a proper ipv6 address?

Comment (Stebalien) [reply net/net.go]: So, there are two cases her:

1. The IP is nil. In that case, we probably *would* want to default to the unspecified address (icky but go does this as well).
2. The IP is not nil but not a valid ipv6 address (the wrong size, that's the only case). In this case, we should probably return nil.

Comment (Stebalien) [reply net/net.go]: Fixed. Thoughts?

**Review by vyzo — APPROVED** [review]
LGTM

File: net/net.go (hunk 1 above)
Comment (vyzo) [reply net/net.go]: yeah, that's sane behaviour.

**Pull request merged**